server/discover.go

```diff
@@ -352,6 +352,10 @@ func (s *Server) entryToRequest(entry *set.Entry, transformer put.PathTransforme
 		r.Meta[put.MetaKeySymlink] = entry.Dest
 	}
 
+	for i, v := range given.Metadata {
+		r.Meta[i] = v
```

Comment:
put package should probably have an "AddUserMetadata()" function that does this loop, but returns an error if key is one of the MetaKey* constants, which only we should be setting. The user supplied key should probably also be put in to our namespace, ie. prefixed with the same prefix as our MetaKey* constants, in a sub-namespace of "user". The checker should allow the user to include the full correct namespace, or prefix it for them if not supplied.

Also, "i" as a loop variable should be reserved for an incrementing integer. Here it would be typical to call it "k" for the key of the map.